examples/medplum-fhircast-demo/src/hooks/index.ts

```diff
@@ -0,0 +1,100 @@
+import { useEffect, useMemo, useRef, useState } from 'react';
+import { FHIRcastMessagePayload, WebSocketMessage, WrappedWebSocket, isWebSocketMessage } from '../utils';
+import { getOrCreateWebSocketClient } from '../websocket-client';
+
+type HubWSConnectionOptions = {
+  onConnect: () => void;
+};
+
+/* eslint-disable-next-line @typescript-eslint/no-redundant-type-constituents */
+type WrappedClientWebSocket = WrappedWebSocket<WebSocket> & {
+  on: (messageType: string, handler: (message: Record<string, any>) => void) => void;
```

Comment:
Let's discuss offline whether we want to use websocket wrappers.  I'd like the understand the benefits of the abstraction vs the complexity.  In particular, I want to make sure we're thinking about this from the perspective of someone coming into this cold with zero context, and how they will process it.

Reply:
Removed the need for them in the latest changes. Should simplify things dramatically.